Lexical editor

akzhysparrow opened the pull request:
- Add editor and related components
- Update configurations to support editor features

justin-ss commented on packages/react-components/rollup.config.js:
```diff
@@ -45,6 +51,31 @@ module.exports = [
       resolve(),
       commonjs(),
+      alias({
+        entries: [
```
@akzhysparrow  can we write a common rollup config and extend those in dev and prod config, instead of repeating this?

justin-ss commented:
@akzhysparrow  can you add docs too?

justin-ss commented on packages/react-components/src/editor/editor.tsx:
```diff
@@ -0,0 +1,240 @@
+} from './utils/findReactChild';
+
+const initialConfig: InitialConfigType = {
+  namespace: 'MyEditor',
```
can we have the namespace as `TwigsEditor` ?

justin-ss commented on packages/react-components/src/editor/editor.tsx:
```diff
@@ -0,0 +1,240 @@
+      }}
+    >
+      <>{showToolbar && <ToolbarPlugin tools={toolbarTools} />}</>
+      <EditorArea />
```
how do we style EditorArea? Using `class: editor-container ` ?

akzhysparrow replied:
Yes, currently we will have to use class name to target the element

justin-ss commented on packages/react-components/src/editor/utils/findReactChild.ts:
file naming convention is not followed for some files

akzhysparrow commented:
> @akzhysparrow can you add story & docs?

Added basic documentation and story

justin-ss commented:
- Styles for heading
- use input for link, instead of textarea
@akzhysparrow

akzhysparrow commented:
> * Styles for heading
> * use input for link, instead of textarea
>   @akzhysparrow

Updated heading styles and changed textarea to input @justin-ss

justin-ss commented:
> @akzhysparrow there should be some space between these items toolbar items in its active state, no? ![image](https://private-user-images.githubusercontent.com/105047491/310792038-39d2bf8e-cc20-46be-93f1-96fb1a984d89.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3OTgzODQsIm5iZiI6MTcwOTc5ODA4NCwicGF0aCI6Ii8xMDUwNDc0OTEvMzEwNzkyMDM4LTM5ZDJiZjhlLWNjMjAtNDZiZS05M2YxLTk2ZmIxYTk4NGQ4OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwN1QwNzU0NDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNmMwM2MzODg1YjkxNDA2NTNkOWRiZGE2MmRhZTg4YzJlZDc1ZDc4ZTFlMTU5MmYyMGZiMWEyYjJiMDQ1NGYwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.8FGeQ6aLtHlWbUpO-2Z_bM_i6pZtxO8X89lVCv3ibh8)

same in toolbar also IMO. Of course we have the ability customize, but still if we package the basic version a bit more neater, that would be nice.
